File: leetcode/380/insertDeleteGetRandom.swift
class RandomizedSet {
    private var lst : [Int] = []

    init() {}
    
    func insert(_ val: Int) -> Bool {
        if lst.contains(val) {
            return false
        }
        else{
            lst.append(val)
            return true
        }
    }
    
    func remove(_ val: Int) -> Bool {
        if let idx = lst.firstIndex(of: val) {
            lst.remove(at: idx)
            return true
        } else {
            return false
        }
    }
    
    func getRandom() -> Int {
        return lst.randomElement()!
        
    }
}

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * let obj = RandomizedSet()
 * let ret_1: Bool = obj.insert(val)
 * let ret_2: Bool = obj.remove(val)
 * let ret_3: Int = obj.getRandom()
 */

File: leetcode/380/insertDeleteGetRandom.py
import random
class RandomizedSet:

    def __init__(self):
        self.tmp = []
        self.dic = {}

    def insert(self, val: int) -> bool:
        if val not in self.dic:
            self.dic[val] = len(self.tmp)
            self.tmp.append(val)
            return True
        else:
            return False

    def remove(self, val: int) -> bool:
        if val in self.dic:
            idx = self.dic[val]
            self.tmp[idx] = self.tmp[-1]
            self.dic[self.tmp[-1]] = idx
            self.tmp.pop()
            del self.dic[val]
            
            return True
        else:
            return False

    def getRandom(self) -> int:
        return random.choice(self.tmp)


# Your RandomizedSet object will be instantiated and called as such:
# obj = RandomizedSet()
# param_1 = obj.insert(val)
# param_2 = obj.remove(val)
# param_3 = obj.getRandom()
